src/snapshot.ts

```diff
@@ -185,18 +118,22 @@ export class Snapshot {
    *   const apiResponse = data[0];
    * });
    */
-  delete(callback) {
+  delete(): Promise<google.protobuf.Empty>;
+  delete(callback: RequestCallback<google.protobuf.Empty>): void;
+  delete(callback?: RequestCallback<google.protobuf.Empty>):
+      void|Promise<google.protobuf.Empty> {
     const reqOpts = {
       snapshot: this.name,
     };
     callback = callback || util.noop;
-    this.parent.request(
-        {
-          client: 'SubscriberClient',
-          method: 'deleteSnapshot',
-          reqOpts,
-        },
-        callback);
+    (this.parent as PubSub)
```

Comment:
Can we do something to remove the cast here? Seems like if we just improved the type of whatever `Subscription#request` is that this wouldn't be needed.